x/gov/codec.proto

```diff
@@ -0,0 +1,93 @@
+syntax = "proto3";
+
+package gov;
+
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "github.com/iov-one/weave/coin/codec.proto";
+import "google/protobuf/timestamp.proto";
+
+// Electorate defines who may vote in an election. This same group can be used in many elections
+// and is stored for re-use
+message Electorate {
+    // Participants defines a list of all signatures that are allowed to participate in a vote
+    repeated Participant participants = 1 [(gogoproto.nullable) = false];
+}
+
+// Participant clubs together a signature with a weight. The greater the weight
+// the greater the power of a participant.
+message Participant {
+    bytes signature = 1 [(gogoproto.casttype) = "github.com/iov-one/weave.Address"];
+    uint32 power = 2 [(gogoproto.casttype) = "Weight"];
+}
+
+// Election Rules defines how an election is run. A proposal must be voted upon via a pre-defined ruleset.
+message ElectionRules {
+    // Human readable title.
+    string title = 1;
+    // Reference to the electorate to define the group of possible voters.
+    bytes electorate_id = 2;
+    // Duration how long the voting period will take place.
+    uint32 voting_period_hours = 3;
+    // Required Vote this is percentage of all eligible voters, not the ones who voted.
+    // (integer in steps of 0.01%, so 10000 is 100%. should disallow anything below 5001)
+    uint32 required_votes = 4;
+}
+
+// A text form proposal for an on-chain governance process.
+message TextProposal {
```

Comment:
The name suggests that there are other types of proposals. Is there anything planned? Otherwise I'd go for TextProposal -> Proposal, since this is also referenced via proposal_id.

Reply:
This is planned.
Super MvP version here and we will need proposals that effect on-chain parameter updates.
This will eventually contain some union type of proposals and the logic to handle them.. Or some way to add that logic. Gotta think a lot on that one

Reply:
The "text proposal" type was mentioned in the original spec. I will rename the `CreateProposal` message to match this name. With new types we can revisit all of this.